spec/controllers/concerns/effective_user_spec.rb

```diff
@@ -22,23 +47,40 @@
       end
     end
 
-    context 'logged out with doc capture session user id' do
+    context 'logged in' do
       before do
-        session[:doc_capture_user_id] = user.id
+        stub_sign_in user
       end
 
       it 'returns session user id' do
         expect(subject).to eq user
       end
-    end
 
-    context 'logged in' do
-      before do
-        stub_sign_in user
+      context 'with valid doc capture session user id that is not the logged-in user' do
+        let(:doc_capture_user) { create(:user) }
+
+        before do
+          session[:doc_capture_user_id] = doc_capture_user.id
+        end
+
+        it 'returns doc capture user id' do
+          expect(subject).to eq(doc_capture_user)
+        end
       end
 
-      it 'returns session user id' do
-        expect(subject).to eq user
+      context 'with invalid doc capture session user id' do
+        before do
+          session[:doc_capture_user_id] = -1
+        end
+
+        it 'returns logged in user' do
+          expect(subject).to eql(user)
+        end
+
+        it 'deletes the session key' do
+          subject
+          expect(session).not_to include(:doc_capture_user_id)
```

Comment:
could use the change syntax if you want:
```suggestion
          expect { subject }.to(change { session[:doc_capture_user_id] }.to(nil))
```